middleware/src/middleware.go

```diff
@@ -164,3 +166,97 @@ func (middleware *Middleware) SampleInfo() rpcmessages.SampleInfoResponse {
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+
+	case rpcmessages.Unmount:
+		log.Println("Executing a USB flashdrive unmount via the cmd script")
+		out, err := middleware.runBBBCmdScript("flashdrive", "unmount")
+		if err != nil {
+			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+
+	default:
+		errorMessage := "Method " + string(args.Method) + " not supported for Flashdrive()."
+		return rpcmessages.GenericResponse{Success: false, Message: errorMessage}, errors.New(errorMessage)
+	}
+}
+
+// Backup returns a GenericResponse struct in response to a rpcserver request
+func (middleware *Middleware) Backup(method rpcmessages.BackupArgs) (rpcmessages.GenericResponse, error) {
+	switch method {
+	case rpcmessages.BackupSysConfig:
+		log.Println("Executing a backup of the system config via the cmd script")
+		out, err := middleware.runBBBCmdScript("backup", "sysconfig")
+		if err != nil {
+			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+
+	case rpcmessages.BackupHSMSecret:
+		log.Println("Executing a backup of the c-lightning hsm_secret via the cmd script")
+		out, err := middleware.runBBBCmdScript("backup", "hsm_secret")
+		if err != nil {
+			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+
+	default:
+		errorMessage := "Method " + string(method) + " not supported for Backup()."
+		return rpcmessages.GenericResponse{Success: false, Message: errorMessage}, errors.New(errorMessage)
+	}
+}
+
+// Restore returns a GenericResponse struct in response to a rpcserver request
+func (middleware *Middleware) Restore(method rpcmessages.RestoreArgs) (rpcmessages.GenericResponse, error) {
+	switch method {
+	case rpcmessages.RestoreSysConfig:
+		log.Println("Executing a restore of the system config via the cmd script")
+		out, err := middleware.runBBBCmdScript("restore", "sysconfig")
+		if err != nil {
+			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+
+	case rpcmessages.RestoreHSMSecret:
+		log.Println("Executing a restore of the c-lightning hsm_secret via the cmd script")
+		out, err := middleware.runBBBCmdScript("restore", "hsm_secret")
+		if err != nil {
+			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+
+	default:
+		errorMessage := "Method " + string(method) + " not supported for Restore()."
+		return rpcmessages.GenericResponse{Success: false, Message: errorMessage}, errors.New(errorMessage)
```

Comment:
As weird as it looks, returning the error twice is actually something I'd support. This way we do not need to unpack the error in the rpc call of the wallet app and the frontend twice.